chore: remove facades and dart-specific patterns

jelbourn opened the pull request:
R: @hansl @kara @robertmesserle

* Replaces `@OneOf` with string literal types.
* Removes facades and uses standard TS / JS APIs
* Uses standard `angular2/testing` imports

This leaves only two things from facades: `BaseException` and `PromiseCompleter`, which don't have a complete drop-in replacement.

hansl commented on src/main.ts:
```diff
@@ -1,13 +1,10 @@
 import {provide} from 'angular2/core';
 import {createOverlayContainer} from './core/overlay/overlay-container';
 
-BrowserDomAdapter.makeCurrent();
```
This isn't needed anymore?

jelbourn replied:
Nay!

hansl commented:
One interrogation, but LGTM!  :shipit:

hansl commented:
Actually, I'm looking at the errors in Travis and these seem legitimate, based on the change from `OneOf(....)` to string enumeration types. Could you take a deeper look?

jelbourn commented:
@hansl all better now; it was mostly that `@OneOf` sneakily also set a default value. Also changed sidenav's `toggle` to use a default param and removed a progress-circle test for a behavior it shouldn't actually exhibit.

jelbourn commented:
Also added a type alias in `Dir` to reduce repetition.

hansl commented on src/core/annotations/field-value.ts:
```diff
@@ -22,13 +19,13 @@ function booleanFieldValueFactory() {
       get() { return this[localKey]; },
       set(value: boolean) {
-        this[localKey] = isPresent(value) && value !== null && String(value) != 'false';
+        this[localKey] = value != null && `${value}` !== 'false';
```
`\`${value}\`` kinda weirds me out. Could you use `toString()` or `'' + ` ? or just `value != 'false'`?

jelbourn replied:
`value != false` isn't allowed by TypeScript (comparing string to boolean).

To me `${value}` is way cleaner than using `toString()` and more readable than `'' + value` (due to the boolean operator) while being the same number of characters).

hansl commented:
Few nits, no blockers. LGTM.

hansl commented:
Also, this means that unless the template compiler verify typings (which I'm not sure it does), the replacement of `OneOf` by native string literal types may lead to undefined behaviour for typos or invalid values.
